core/runtime/snapshot_util.rs

```diff
@@ -232,11 +258,22 @@ pub(crate) fn set_snapshotted_data(
     scope.add_context_data(local_context, local_handled_promise_rejection_cb);
   assert_eq!(offset, JS_HANDLED_PROMISE_REJECTION_CB_DATA_INDEX);
 
+  let mut ext_source_maps =
+    Vec::with_capacity(snapshotted_data.ext_source_maps.len() * 2);
+  for (key, value) in snapshotted_data.ext_source_maps.iter() {
+    let value = std::str::from_utf8(value).unwrap();
```

Comment:
Is there a better way to do this? Maybe using ArrayBuffers instead of converting to a string?

Reply:
I think that we should switch to using explicit `serde_v8` serialization for the snapshot. I've added some benchmarks so we can run some experiments here, just haven't had time to experiment with it.

For now, we can probably land with strings and then re-address all of the snapshot sidecar data.